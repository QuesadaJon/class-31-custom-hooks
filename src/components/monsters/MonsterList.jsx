import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Monster from './Monster';

const MonsterList = ({ monsters }) => {
  const monsterElements = monsters.map((monster) => (
    <li key={monster.name}>
      <Link to={`/monster/${monster.name}`}>
        <Monster
          name={monster.name}
          image={monster.image}
        />
      </Link>
    </li>
  ));
  return <ul data-testid="monsters">{monsterElements}</ul>;
};

MonsterList.propTypes = {
  monsters: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    })
  ).isRequired
};

export default MonsterList;
