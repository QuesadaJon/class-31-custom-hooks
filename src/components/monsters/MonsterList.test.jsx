import React from 'react';
import { render } from '@testing-library/react';
import MonsterList from './MonsterList';
import { MemoryRouter } from 'react-router-dom';

const monsters = [
  {
    image: 'https://image.com',
    name: 'name',
  },
  {
    image: 'https://image.com',
    name: 'bob',
  },
  {
    image: 'https://image.com',
    name: 'sarah',
  }
];

describe('MonsterList component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <MonsterList monsters={monsters} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
})
;
