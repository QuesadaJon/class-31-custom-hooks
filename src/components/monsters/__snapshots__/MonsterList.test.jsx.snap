// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonsterList component matches a snapshot 1`] = `
<DocumentFragment>
  <ul
    data-testid="monsters"
  >
    <li>
      <a
        href="/monster/name"
      >
        <figure>
          <p>
            name
          </p>
          <img
            alt="name"
            src="https://image.com"
          />
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/monster/bob"
      >
        <figure>
          <p>
            bob
          </p>
          <img
            alt="bob"
            src="https://image.com"
          />
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/monster/sarah"
      >
        <figure>
          <p>
            sarah
          </p>
          <img
            alt="sarah"
            src="https://image.com"
          />
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
